test/test_bmark.jl

```diff
@@ -65,6 +75,57 @@ function test_bmark()
     pretty_stats(stats[:dummy])
   end
 
+  if Threads.nthreads() > 1  # if we have multi_thread
+    @testset "Multithread vs Single-Thread Consistency" begin
+      problems = [
+        ADNLPModel(x -> sum(x .^ 2), ones(2), name = "Quadratic"),
+        ADNLPModel(
+          x -> sum(x .^ 2),
+          ones(2),
+          x -> [sum(x) - 1],
+          [0.0],
+          [0.0],
+          name = "Cons quadratic",
+        ),
+        ADNLPModel(
+          x -> (x[1] - 1)^2 + 4 * (x[2] - x[1]^2)^2,
+          ones(2),
+          x -> [x[1]^2 + x[2]^2 - 1],
+          [0.0],
+          [0.0],
+          name = "Cons Rosen",
+        ),
+      ]
+      callable = CallableSolver()
+      solvers = Dict(
+        :dummy => dummy_solver,
+        :callable => callable,
+        :dummy_solver_specific =>
+          nlp -> dummy_solver(
+            nlp,
+            callback = (nlp, solver, stats) -> set_solver_specific!(stats, :foo, 1),
+          ),
+      )
+
+      # Run the single-threaded version
+      single_threaded_result = bmark_solvers_signle_thread(solvers, problems)
+      multithreaded_result = bmark_solvers(solvers, problems)
+
+      # Compare the results
+      @test length(single_threaded_result) == length(multithreaded_result)
+
+      for key in keys(single_threaded_result)
+        @test single_threaded_result[key][1, :status] == multithreaded_result[key][1, :status]
+        @test single_threaded_result[key][2, :status] == multithreaded_result[key][2, :status]
+        @test single_threaded_result[key][3, :status] == multithreaded_result[key][3, :status]
+
+        @test single_threaded_result[key][1, :name] == multithreaded_result[key][1, :name]
+        @test single_threaded_result[key][2, :name] == multithreaded_result[key][2, :name]
+        @test single_threaded_result[key][3, :name] == multithreaded_result[key][3, :name]
```

Comment:
Could you replace these with loops over the rows of the dataframes?

You could also check the solver-specific info.

Reply:
I suggest we actually compare the dataframes directly (or approximate ),